var/spack/repos/builtin/packages/openspeedshop-utils/package.py

```diff
@@ -0,0 +1,343 @@
+
+        cmakeOptions.extend(BaseOptions)
+
+    def set_mpi_cmakeOptions(self, spec, cmakeOptions):
+        # Appends to cmakeOptions the options that will enable
+        # the appropriate MPI implementations
+
+        MPIOptions = []
+
+        # openmpi
+        if spec.satisfies('+openmpi'):
+            MPIOptions.append('-DOPENMPI_DIR=%s' % spec['openmpi'].prefix)
+        # mpich
+        if spec.satisfies('+mpich'):
+            MPIOptions.append('-DMPICH_DIR=%s' % spec['mpich'].prefix)
+        # mpich2
+        if spec.satisfies('+mpich2'):
+            MPIOptions.append('-DMPICH2_DIR=%s' % spec['mpich2'].prefix)
+        # mvapich
+        if spec.satisfies('+mvapich'):
+            MPIOptions.append('-DMVAPICH_DIR=%s' % spec['mvapich'].prefix)
+        # mvapich2
+        if spec.satisfies('+mvapich2'):
+            MPIOptions.append('-DMVAPICH2_DIR=%s' % spec['mvapich2'].prefix)
+        # mpt
+        if spec.satisfies('+mpt'):
+            MPIOptions.append('-DMPT_DIR=%s' % spec['mpt'].prefix)
+
+        cmakeOptions.extend(MPIOptions)
+
+    def setup_environment(self, spack_env, run_env):
+        """Set up the compile and runtime environments for a package."""
+
+        # Common settings to both offline and cbtf versions
+        # of OpenSpeedShop
+        run_env.prepend_path('PATH', self.prefix.bin)
+
+        # Find Dyninst library path, this is needed to
+        # set the DYNINSTAPI_RT_LIB library which is
+        # required for OpenSpeedShop to find loop level
+        # performance information
+        dyninst_libdir = find_libraries('libdyninstAPI_RT',
+                                        root=self.spec['dyninst'].prefix,
+                                        shared=True, recursive=True)
+
+        # Set Dyninst RT library path to support OSS loop resolution code
+        run_env.set('DYNINSTAPI_RT_LIB', dyninst_libdir)
+
+        # Find openspeedshop library path
+        oss_libdir = find_libraries('libopenss-framework',
+                                    root=self.spec['openspeedshop-utils'].prefix,
+                                    shared=True, recursive=True)
+        run_env.prepend_path('LD_LIBRARY_PATH',
+                             os.path.dirname(oss_libdir.joined()))
+
+        run_env.set('OPENSS_RAWDATA_DIR', '.')
+
+        cbtf_mc = '/sbin/cbtf_mrnet_commnode'
+        cbtf_lmb = '/sbin/cbtf_libcbtf_mrnet_backend'
+        run_env.set('XPLAT_RSH', 'ssh')
+        run_env.set('MRNET_COMM_PATH',
+                    join_path(self.spec['cbtf-krell'].prefix + cbtf_mc))
+        run_env.set('CBTF_MRNET_BACKEND_PATH',
+                    join_path(self.spec['cbtf-krell'].prefix + cbtf_lmb))
+        run_env.prepend_path('PATH', self.spec['mrnet'].prefix.bin)
+        run_env.prepend_path('PATH', self.spec['cbtf-krell'].prefix.bin)
+        run_env.prepend_path('PATH', self.spec['cbtf-krell'].prefix.sbin)
+        run_env.prepend_path('PATH', self.spec['python'].prefix.bin)
```

Comment:
Most Spack packages don't add dependencies to their module files in order to prevent things from being polluted.

Reply:
I was having problems executing openspeedshop, so I added these.   I added the python one recently because without it I was getting an error related to python.  I either had to manually load the python module or add this to the module file.

Reply:
Because when I user loads a openspeedshop module outside of spack, all dependencies are taken care of.  Having to load another module hinders usuage.   Plus the user wouldn't know what module to load.

Reply:
when a user